app/presenters/katello/host_package_presenter.rb

```diff
@@ -0,0 +1,21 @@
+module Katello
+  class HostPackagePresenter < SimpleDelegator
+    attr_accessor :installed_package, :upgradable_version
+
+    def initialize(installed_package, upgradables)
+      @installed_package = installed_package
+      @upgradable_version = upgradables[installed_package.name]&.first&.nvra || ''
+      super(@installed_package)
+    end
+
+    def self.with_latest(packages, host)
+      upgradable_packages_map = ::Katello::Rpm.
+                                installable_for_hosts([host]).
+                                select(:name, :nvra, :evr).
+                                order(evr: :desc).
+                                all.
+                                group_by(&:name)
+      packages.map { |p| HostPackagePresenter.new(p, upgradable_packages_map) }
```

Comment:
It seems unnecessary that every instance of HostPackagePresenter needs the entire upgradeable_packages_map - what do you think about moving the lookup into this method and passing just the upgradeable_package itself as the parameter?